### PR TITLE
ipatests: CLI validation of ipa-healthcheck command

### PR DESCRIPTION
**Pull request by rcritten**
Test for illegal input values.

There is one xfail, BZ reported.

### CODE REVIEW

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 1]
```diff
@@ -1450,3 +1451,139 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+            cmd = self.base_cmd + ["--indent", option]
+            result = self.master.run_command(cmd, raiseonerr=False)
+            assert result.returncode == 2
+            assert 'invalid int value' in result.stderr_text
+
+        # unusual success, arguably odd but not invalid :-)
+        for option in ('-1', '5000'):
+            cmd = self.base_cmd + ["--indent", option]
+            result = self.master.run_command(cmd, raiseonerr=False)
+            assert result.returncode == 0
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1493]: `rasieonerr=False` can be removed so the assert statement. If return code is not 0, command itself will fail.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 2]
```diff
@@ -1450,3 +1451,139 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+
+    def test_severity(self):
+        """
+        Valid and invalid --severity
+        """
+        # Baseline, there should be no errors
+        cmd = ["ipa-healthcheck", "--severity", "SUCCESS"]
+        result = self.master.run_command(cmd, raiseonerr=False)
+        assert result.returncode == 0
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1502]: `rasieonerr=False` can be removed so the assert statement. If return code is not 0, command itself will fail.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 3]
```diff
@@ -1450,3 +1451,139 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+        data = json.loads(result.stdout_text)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+
+        # All the other's should return nothing
+        for severity in ('WARNING', 'ERROR', 'CRITICAL'):
+            cmd = ["ipa-healthcheck", "--severity", severity]
+            result = self.master.run_command(cmd, raiseonerr=False)
+            assert result.returncode == 0
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1511]: `rasieonerr=False` can be removed so the assert statement. If return code is not 0, command itself will fail.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 4]
```diff
@@ -1450,3 +1451,139 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+        Verify the --input-file option
+        """
+        # ipa-healthcheck overwrites output file, no need to generate
+        # a randomized name.
+        outfile = "/tmp/healthcheck.out"
+
+        # create our output file
+        cmd = ["ipa-healthcheck", "--output-file", outfile]
+        result = self.master.run_command(cmd, raiseonerr=False)
+        assert result.returncode == 0
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1533]: `rasieonerr=False` can be removed so the assert statement. If return code is not 0, command itself will fail.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 5]
```diff
@@ -1450,3 +1451,139 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+
+        # load the file
+        cmd = ["ipa-healthcheck", "--failures-only", "--input-file", outfile]
+        result = self.master.run_command(cmd, raiseonerr=False)
+        assert result.returncode == 0
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1538]: `rasieonerr=False` can be removed so the assert statement. If return code is not 0, command itself will fail.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 6]
```diff
@@ -1450,3 +1451,139 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+
+    def test_pki_healthcheck(self):
+        """
+        Ensure compatibility with pki-healthcheck
+
+        Running on a clean system should produce no errors. This will
+        ensure ABI compatibility.
+        """
+        result = self.master.run_command(["pki-healthcheck"], raiseonerr=False)
+        assert result.returncode == 0
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1589]: `rasieonerr=False` can be removed so the assert statement. If return code is not 0, command itself will fail.

**Comment by rcritten** [comment]
Changes applied

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 7]
```diff
@@ -1450,3 +1451,146 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
...
+        """
+        Verify that when arguments are specified to --list-sources
+        option, error is displayed on the console.
+        """
+        cmd = self.base_cmd + ["--list-sources", "source"]
+        result = self.master.run_command(cmd, raiseonerr=False)
+        assert result.returncode == 2
+        assert (
+            "ipa-healthcheck: error: unrecognized arguments: source"
+            in result.stdout_text
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:1595]: Error would be part of `result.stderr_text`

Comment (kaleemsiddiqu) [reply ipatests/test_integration/test_ipahealthcheck.py]: Done

**Comment by rcritten** [comment]
Dropped temp commit and rebased to master.

**Comment by flo-renaud** [comment]
ACK'ing as tests are green and @mrizwan93 already reviewed the code.

**Comment by rcritten** [comment]
master:

* c5853768a778069945237e0a039715d89e16a94e ipatests: CLI validation of ipa-healthcheck command
* 143dea18fe9bc2ebc65a8dbdcfa44d331adc161c Added negative test case for --list-sources option

**Pull request closed without merging**